yt_dlp/extractor/thisoldhouse.py

```diff
@@ -41,7 +46,12 @@ class ThisOldHouseIE(InfoExtractor):
     def _real_extract(self, url):
         display_id = self._match_id(url)
         webpage = self._download_webpage(url, display_id)
-        video_id = self._search_regex(
-            r'<iframe[^>]+src=[\'"](?:https?:)?//(?:www\.)?thisoldhouse\.(?:chorus\.build|com)/videos/zype/([0-9a-f]{24})',
-            webpage, 'video id')
+        if 'To Unlock This content' in webpage:
+            self.raise_login_required(method='cookies')
+        video_url = self._search_regex(
+            r'<iframe[^>]+src=[\'"]((?:https?:)?//(?:www\.)?thisoldhouse\.(?:chorus\.build|com)/videos/zype/([0-9a-f]{24})[^\'"]*)[\'"]',
+            webpage, 'video url')
+        if 'subscription_required=true' in video_url:
+            return self.url_result(self._request_webpage(video_url, display_id).url, 'Zype', display_id)
```

Comment:
```suggestion
            return self.url_result(self._request_webpage(HEADRequest(video_url), display_id).geturl(), 'Zype', display_id)
```

or 
```suggestion
            return self.url_result(video_url, 'Generic', display_id)
```
should work too

Reply:
i don't want to pass to generic bcoz it's bloated.
What are the advantages of the first one?

Reply:
`.geturl()` is the right way to get redirect URL

HeadRequest makes it so the actual webpage isn't downloaded, only the headers. Check that the website doesnt block HEAD requests though

> i don't want to pass to generic bcoz it's bloated.

redirect is the first thing GenericIE tries
```py
        self.to_screen('%s: Requesting header' % video_id)

        head_req = HEADRequest(url)
        head_response = self._request_webpage(
            head_req, video_id,
            note=False, errnote='Could not send HEAD request to %s' % url,
            fatal=False)

        if head_response is not False:
            # Check for redirect
            new_url = head_response.geturl()
            if url != new_url:
                self.report_following_redirect(new_url)
                if force_videoid:
                    new_url = smuggle_url(
                        new_url, {'force_videoid': force_videoid})
                return self.url_result(new_url)
```

but your wish. Either method is fine

Reply:
I should have been more verbose...
What i meant with `bcoz it's bloated` is that passing to generic prints extra stuff the cli which makes the logs bloated.
I applied the first suggestion and site indeed accepts Head request so thanks :)